tc/sgx/trusted_worker_manager/enclave/enclave_data.cpp

```diff
@@ -93,35 +107,55 @@ void EnclaveData::DeserializeSealedData(const std::string& inSerializedEnclaveDa
 
     // Public signing key
     pvalue = json_object_dotget_string(keystore_object, "SigningKey.PublicKey");
+    pvalue_len = strlen(pvalue);
     tcf::error::ThrowIf<tcf::error::ValueError>(
         !pvalue, "Failed to retrieve public signing key from the key store");
 
     svalue.assign(pvalue);
     public_signing_key_.Deserialize(svalue);
 
+    // Clear local variables storing secrets before re-assigning new value
+    svalue.clear();
+    memset_s((char*)pvalue, pvalue_len, 0, pvalue_len);
```

Comment:
Max size to set in memset_s will be size of key

Reply:
We don't have max size of key like public signing key or public encryption key.

Reply:
Use Max expected size of public key. memset_s will check if pvalue_len < MAX_SIZE and will not result in any buffer overflow while copying.

Reply:
I understand but in Avalon we haven't defined max size of public key.

Reply:
FYI, in the future, with quantum-computing resistant crypto, the public key may be quite large.

Reply:
@manju956 define a MAX_KEY_SIZE based on your current implementation and check that buffer size is less than MAX_SIZE.

Reply:
I could compute strlen upto size of serialized encryption/signing keys as the keys stored in these local variables are infact serialized version.

Reply:
these are serialized version of keys, so i am not sure how to define MAX_KEY_SIZE for different keys.